packages/compass-generative-ai/src/atlas-ai-service.ts

```diff
@@ -298,7 +298,39 @@ export class AtlasAiService {
     return body;
   }
 
+  async _e2eTestOnlyOverrideAIFeaturePreferences() {
+    try {
+      // When we're running e2e tests and want to customize these preferences on the fly
+      // we make a request to the server to override the preferences.
+      const {
+        enableGenAIFeaturesAtlasProject,
+        enableGenAISampleDocumentPassingOnAtlasProject,
+        enableGenAIFeaturesAtlasOrg,
+        optInDataExplorerGenAIFeatures,
+      } = await fetch('e2e-test-atlas-preferences-override').then((res) =>
```

Comment:
Feels a bit hacky to dynamically make a request to a local server via the proxy here, but I'm not sure there's a more elegant way. Open to thoughts folks may have here.

Reply:
If this is based on preferences, I would rather make it so that e2e tests in web can modify preferences similar to the desktop e2e tests and not add extra endpoints and e2e test logic directly to the code here